Allow user to select starting level instead of defaulting to 5

AliHassan245 opened the pull request:
Previously, the game always initialized at level 5, ignoring user selection. This fix removes the hardcoded level initialization, enabling users to choose their desired starting level.

AliHassan245 commented:
@quozl @chimosky please have a look.

quozl commented:
Reviewed.  Looks wrong in a few ways, here's a list to start with;

* [x] rewrite your commit message, see https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits
* [x] don't change lines unrelated to your purpose.

AliHassan245 commented:
I’ve updated the commit message on the pull request and reverted back the lines that were not relevant to the issue. Feel free to check it out now!

chimosky commented on BlockParty.py:
```diff
@@ -262,7 +263,7 @@ def key_action(self, key):
                     self.queue_draw_complete()
                     self.next_tick = time.time() + self.time_step
-                    self.game_mode = self.PLAY
+                    self.game_mode = self.PLAY  
```
Extra spaces at the end, which would fail flake8 tests.

Please look at the diffs of your change before making a commit.

chimosky commented on BlockParty.py:
```diff
@@ -202,7 +202,8 @@ def init_game(self):
         self.score = 0
         self.new_figure()
-        self.set_level(5)
+        if not hasattr(self, 'level') or self.level is None:
+            self.set_level(0)
```
There's no need for the if statement here because this is the first place the level is set when the game starts.

AliHassan245 replied:
Yep, it can also work without if statement. I just added it for additional check.

quozl replied:
No reason for additional check though.

chimosky commented:
Your commit message is difficult to understand, "Set initial game level to zero" is better.

quozl commented:
Again;
* [ ] commit message inadequate,

While we could rewrite the commit message ourselves, you should do it to make sure that your intention for the change is fully understood when git history is used.  Be sure not to reference issue numbers, because these are not kept in git.

Now, looking at the actual change you're proposing, I disagree with it, because we've been here before and the reasons you give (none at all) are not good enough for me.

6eadba7b5b4f9e9ec193b7aef4416aa48a720f3b changed the level from 0 to 5 and we accepted it then; why should we change it now?  Why did you not reference this change in your commit message?

AliHassan245 commented:
Yes, previously the game defaulted to level 5, which meant it started with a higher difficulty, blocks were falling faster right from the beginning. However, in most games, the default starting level is 0, allowing the difficulty to scale gradually.

Commit:
Set intial game level to 0

-Previously the commit 6eadba7 introduced a regression where the game always initialized at level 5, starting with higher difficulty as blocks fall faster at higher levels.
- This update sets the initial level to 0 so that difficulty increases progressively with gameplay.

AliHassan245 commented:
I have updated the commit messages. This PR is all set and ready to go.🚀

quozl commented:
So, you're argument here in the pull request is that the game has regressed with 6eadba7b5b4f9e9ec193b7aef4416aa48a720f3b; please add that information to the commit message.  That's what we mean by "when the problem was introduced by a previous commit, mention the hash;"

Please try to comply with commit message guidelines; we have developed them over years to support the use of git, not GItHub.  When you miss one of the guidelines, that makes us wonder why.

AliHassan245 commented:
Thanks for the feedback! I’ve updated the commit message to follow the Sugar Labs guidelines.

AliHassan245 commented:
@quozl  @chimosky This PR is all set and ready to go 🚀😊
